brian2/core/network.py

```diff
@@ -487,11 +516,12 @@ def remove(self, *objs):
                                     "objects from Network")
 
     def _full_state(self):
+        all_objects = self.get_all_objects()
         state = {}
-        for obj in self.objects:
+        for obj in all_objects:
             if hasattr(obj, '_full_state'):
                 state[obj.name] = obj._full_state()
-        clocks = set([obj.clock for obj in self.objects])
+        clocks = set([obj.clock for obj in all_objects])
```

Comment:
Minor stylistic point (and this also applies to the code before your change), this would be nicer when written as:
```Python
clocks = {obj.clock for obj in all_objects}
```